### PR TITLE
PXB-2759 regression on check_for_passwords.sh

### PR DESCRIPTION
**Pull request by rahulmalik87**
https://jira.percona.com/browse/PXB-2759

added sleep 3 befor checking output of ps

### CODE REVIEW

**Comment by rahulmalik87** [comment]
jenkin results https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-multijob/182

**Comment by rahulmalik87** [comment]
jenkin results. https://pxb.cd.percona.com/job/percona-xtrabackup-8.0-test-param/201/

**Review by satya-bodapati — APPROVED** [review]
Please squash before the merge

File: storage/innobase/xtrabackup/src/backup_copy.cc [hunk 1]
```diff
@@ -2639,6 +2639,8 @@ bool decrypt_decompress() {
   ret = run_data_threads(".", decrypt_decompress_thread_func,
                          xtrabackup_parallel, "decrypt and decompress");
 
+  debug_sync_point("decrypt_decompress_func");
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/src/backup_copy.cc:2642]: Do we really need this one? checking for `decrypt` isn't enough?

File: storage/innobase/xtrabackup/test/t/check_for_passwords_debug.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+function grep_in_datadir_ps_decrypt() {
+
+  additional_options=$1
+
+  xtrabackup --backup $additional_options --target-dir=$topdir/backup
+
+  $XB_BIN $XB_ARGS --decrypt=AES256 $additional_options \
+  --target-dir=$topdir/backup \
+           --debug-sync="decrypt_decompress_func" \
+       2> >( tee $topdir/decompress.log)&
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/test/t/check_for_passwords_debug.sh]: decrypt.log

File: storage/innobase/xtrabackup/test/t/check_for_passwords.sh [hunk 3]
```diff
@@ -70,32 +72,15 @@ function grep_in_datadir_ps() {
...
-function grep_in_datadir_ps_decrypt() {
+function grep_in_datadir_decrypt() {
 
   additional_options=$1
 
   xtrabackup --backup $additional_options --target-dir=$topdir/backup
 
   $XB_BIN $XB_ARGS --decrypt=AES256 $additional_options \
   --target-dir=$topdir/backup \
-       2> >( tee $topdir/pxb_2740.log)&
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/test/t/check_for_passwords.sh:81]: I would leave this part with extra information to validate why this is failing.

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]
@rahulmalik87 although the suggested change might work, I would investigate why this is failing without the debug sync.
Based on your changes, the issue comes from `decrypting` part. I would enhance debugging there to validate why it fails, Having the `decrypting` string on pxb output log should be enough to be passed the argument handling part, which should be stripping sensitive information.

**Comment by rahulmalik87** [comment]
> @rahulmalik87 although the suggested change might work, I would investigate why this is failing without the debug sync.
> Based on your changes, the issue comes from `decrypting` part. I would enhance debugging there to validate why it fails, Having the `decrypting` string on pxb output log should be enough to be passed the argument handling part, which should be stripping sensitive information.

It fails because we don't have any mechanism to ensure decrypting is not finished at the time we are running ps command.
  
Having the `decrypting` string is enough but how we would ensure that decrypting is still running. That's why I thought to run only debug test case to 100% process would be running by the time we run ps -ef

File: storage/innobase/xtrabackup/src/backup_copy.cc (hunk 1 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/backup_copy.cc:2642]: yeah, It is ensure the decrypting is not complete.

File: storage/innobase/xtrabackup/test/t/check_for_passwords_debug.sh (hunk 2 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/test/t/check_for_passwords_debug.sh]: fixed

File: storage/innobase/xtrabackup/test/t/check_for_passwords.sh (hunk 3 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/test/t/check_for_passwords.sh:81]: I can't find a way to ensure the process would be blocked untill we run ps

**Review by altmannmarcelo — APPROVED** [review]
LGTM

**Pull request merged**